src/main/java/io/cdap/plugin/successfactors/source/transport/SuccessFactorsUrlContainer.java

```diff
@@ -118,23 +119,36 @@ private HttpUrl.Builder buildQueryOptions(HttpUrl.Builder urlBuilder, Boolean is
     if (SuccessFactorsUtil.isNotNullOrEmpty(pluginConfig.getFilterOption())) {
       urlBuilder.addQueryParameter(FILTER_OPTION, pluginConfig.getFilterOption());
     }
+
+    boolean isExpandFieldValue = false;
     if (SuccessFactorsUtil.isNotNullOrEmpty(pluginConfig.getSelectOption())) {
-      urlBuilder.addQueryParameter(SELECT_OPTION, pluginConfig.getSelectOption());
-    } else if (isDataFetch) {
+      if (SuccessFactorsUtil.isNotNullOrEmpty(pluginConfig.getExpandOption())) {
+        String selectFieldValue = pluginConfig.getSelectOption().concat(PROPERTY_SEPARATOR)
+          .concat(pluginConfig.getExpandOption());
+        urlBuilder.addQueryParameter(SELECT_OPTION, selectFieldValue);
+        urlBuilder.addQueryParameter(EXPAND_OPTION, pluginConfig.getExpandOption());
+        isExpandFieldValue = true;
+      } else {
+        urlBuilder.addQueryParameter(SELECT_OPTION, pluginConfig.getSelectOption());
+      }
+    } else if (getExpandLevel() <= 1) {
       SuccessFactorsService successFactorsService = SuccessFactorsUtil.getSuccessFactorsService(pluginConfig);
       try {
-        StringBuilder selectFieldValue = new StringBuilder(String.join(PROPERTY_SEPARATOR, successFactorsService.
+        StringBuilder selectNonNav = new StringBuilder(String.join(PROPERTY_SEPARATOR, successFactorsService.
           getNonNavigationalProperties()));
         if (SuccessFactorsUtil.isNotNullOrEmpty(pluginConfig.getExpandOption())) {
-          selectFieldValue.append(PROPERTY_SEPARATOR).append(pluginConfig.getExpandOption());
+          selectNonNav.append(PROPERTY_SEPARATOR).append(pluginConfig.getExpandOption());
+          urlBuilder.addQueryParameter(EXPAND_OPTION, pluginConfig.getExpandOption());
+          isExpandFieldValue = true;
         }
-        urlBuilder.addQueryParameter(SELECT_OPTION, selectFieldValue.toString());
+        urlBuilder.addQueryParameter(SELECT_OPTION, selectNonNav.toString());
 
       } catch (TransportException | SuccessFactorsServiceException | EdmException e) {
-        e.printStackTrace();
+        LOG.error(String.valueOf(e));
```

Comment:
Add some context the error message.

Reply:
Updated the message 5ba462f6e79c0757c5cd2a51632dc0219ed2a944